internal/faustjs.org/docs/gutenberg/tutorial/create-a-custom-block.mdx

```diff
@@ -0,0 +1,248 @@
+        color
+        content
+        message
+      }
+    }
+  `,
+};
+```
+We chose to attach the fragment as a property of the function component itself. This follows best practice for creating [colocated fragments](https://www.apollographql.com/docs/react/data/fragments/#creating-colocated-fragments).
+
+Before moving forward, check to ensure your block is being queried correctly in the GraphiQL IDE. You may need to use your browser's search and find method to search all the possible names under contentBlocks.
+
+<a href="/docs/img/graphiql-query-composer.png" target="_blank">
+  <img
+    src="/docs/img/graphiql-query-composer.png"
+    alt="The GraphiQL IDE is open to the Query Composer, which shows under posts, then nodes, then contentBlocks that the CreateMycoolblock is present, with a selection of attributes visible such as color, content, and message."
+  />
+</a>
+
+### 4.4 Include the block fragment into the page query string
+
+You want to include the fragment in the page query string and reference the block fragment key so that when the page resolves, the payload includes the data from that block.
+
+In this example we will use it in the single page template provided in the Faust Scaffold. You can learn more about the Faust Template Hierarchy [here](/docs/templates).
+
+```jsx {1,7,18} title="wp-templates/single.js"
+import blocks from '../wp-blocks';
+export default function Component(props) {
+...
+}
+Component.query = gql`
+...
+${blocks.MyCoolBlock.fragments.entry}
+query GetPost(
+...
+	post(id: $databaseId, idType: DATABASE_ID, asPreview: $asPreview) {
+		...
+		contentBlocks {
+			name
+			__typename
+			renderedHtml
+			id: nodeId
+			parentId
+			...${blocks.MyCoolBlock.fragments.key}
+      	}
+	}
+...
+}
+```
+
+If you navigate to the page that contains this block (here, named `localhost:3000/my-cool-block`), you should be able to inspect the properties in the console.
+
+<a href="/docs/img/contentBlocks-in-devtools.png" target="_blank">
+  <img
+    src="/docs/img/contentBlocks-in-devtools.png"
+    alt="Preview of the MyCoolBlock in the console."
+  />
+</a>
+
+## 5. Porting over the custom block's style.scss file
+
+While creating your custom block using `@wordpress/create-block`, the `style.scss` file was autogenerated in your block's plugin folder. You can include these in the global styles section of your Decoupled front-end app too, if you wish:
+
+```scss title="/MERL-742-Tutorial/styles/globals.css"
+...
+
+@font-face {
+	font-family: Gilbert;
+	src: url(/app/public/wp-content/plugins/mycoolblock/assets/gilbert-color.otf);
+	font-weight: 700;
+}
+
+.wp-block-create-block-mycoolblock {
+	font-family: Gilbert, sans-serif;
+	font-size: 64px;
+}
+```
+
+Now that you have the styles ported over, you can navigate into the Headless site's page and check that the block matches the styling you wish. Make sure to tweak the block in the editor and verify that the changes you want are being reflected here, too.
+
```

Comment:
I think there is a missing step here. The implementation of the block in headless. Ideally it should match the 

```jsx title="wp-blocks/MyCoolBlock.js"
export default function MyCoolBlock({attributes}) {
  return <p>{ `The color selected is ${attributes.color}! My favorite color: ${attributes.message}!` }
  <h4>{attributes.content}</h4></p>

<p { ...useBlockProps.save() }>
			{ `The color selected is ${attributes.color}! My favorite color: ${attributes.message}!` }
            <RichText.Content tagName="h4" value={ attributes.content }/>
        </p>
}
```

Reply:
I have addressed this item, but I'm afraid my understanding of how to test to ensure proper implementation is still limited. I think I am experiencing the same issue that @matthewguywright discussed during standup- I'm not sure where I should look to debug as I attempt to adapt the block to its WordPress version. Where should I be working? I am currently using `localhost:3000/my-cool-block`, but that does not reflect my changes.

Reply:
Hi @theodesp, thank you for this feedback! I am importing the `WordPressBlocksViewer,` but I haven't grabbed the `flatListToHierarchical` from our example project to introduce to this one (Im working off faust-scaffold). Here is what I have in `single.js` at the moment: 

https://github.com/TeresaGobble/MERL-742-Tutorial/blob/main/wp-templates/single.js